### PR TITLE
Features/run as account

### PR DESCRIPTION
**Pull request by skaue**
Removed dependency to Credentials and made the script run using RunAsAccounts. Adjustments to the code accessing tags.

### CODE REVIEW

**Review by skaue — COMMENTED** [review]
This code removes code from the master branch, rendering it incompatible with the original code.

File: Assert-AutoShutdownSchedule.ps1 [hunk 1]
```diff
@@ -256,104 +253,44 @@ try
...
+    $Conn = Get-AutomationConnection -Name AzureRunAsConnection
+    $resourceManagerContext = Add-AzureRMAccount -ServicePrincipal -Tenant $Conn.TenantID -ApplicationId $Conn.ApplicationID -CertificateThumbprint $Conn.CertificateThumbprint
+    
     $resourceList = @()
     # Get a list of all supported resources in subscription
     $ResourceProcessors | % {
-      Write-Output ('Looking for resources of type {0}' -f $_.ResourceType)
-      $resourceList += @(Find-AzureRmResource -ResourceType $_.ResourceType)
+        Write-Output ('Looking for resources of type {0}' -f $_.ResourceType)
+        $resourceList += @(Get-AzureRmResource -ResourceType $_.ResourceType)
```
Comment (slapointe) [inline Assert-AutoShutdownSchedule.ps1]: @skaue Please use ```Get-AzureRmResource -ODataQuery ("`$filter=resourcetype eq '{0}'" -f $_.ResourceType)```. Get-AzureRmResource -ResourceType have compatibility issues over differents versions

**Review by slapointe — CHANGES_REQUESTED** [review]
Please fix the `Get-AzureRmResource -ResourceType` calls like in the following article: 
https://www.codeisahighway.com/how-to-safely-replace-find-azurermresource-resourcetype-calls-in-azure-powershell-6-x/

**Comment by skaue** [comment]
Like this? :)

**Review by slapointe — APPROVED** [review]

**Comment by slapointe** [comment]
Thanks @skaue for your contribution!

**Pull request merged**